File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        char x = s.charAt(0);
        int cnt = 0;
        int diff = 0;
        
        for(int i = 0; i < s.length(); i++){
            char ch = s.charAt(i);
            
            if(x == ch){
                cnt++;
            } else{
                diff++;
            }
            
            if(cnt == diff){
                answer++;
                x = i < s.length() - 1 ? s.charAt(i+1) : ' ';
                cnt = 0;
                diff = 0;
            }
        }
        
        return answer = cnt == diff ? answer : answer + 1;
    }
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    var answer = 1;
    let ch = s[0];
    let count = 0;
    let flag;
    for (let i = 0; i < s.length; i++){
        if (count == 0 && flag){
            ch = s[i];
            answer++;
            flag = false;
        }
        if (s[i] == ch) count++;
        else count--;
        if (count == 0) flag = true;
    }
    return answer;
}